lib/trip.rb

```diff
@@ -13,28 +14,42 @@ def initialize(
           start_time:,
           end_time:,
           cost: nil,
-          rating:
+          rating:,
+          driver_id: nil,
+          driver: nil
         )
       super(id)
 
       if passenger
         @passenger = passenger
         @passenger_id = passenger.id
-
       elsif passenger_id
         @passenger_id = passenger_id
-
       else
         raise ArgumentError, 'Passenger or passenger_id is required'
       end
 
       @start_time = start_time
-      @end_time = end_time
       @cost = cost
-      @rating = rating
+      @driver_id = driver_id
+      @driver = driver
 
-      if @rating > 5 || @rating < 1
-        raise ArgumentError.new("Invalid rating #{@rating}")
+      if rating.nil?
+        @rating = rating
+      elsif rating > 5 || rating < 1
+        raise ArgumentError.new("Invalid rating #{rating}")
+      else @rating = rating
+      end
```

Comment:
Love the validations! One way to refactor this code could be to make little helper methods for validating each parameter so that the constructor doesn't get cluttered.